lib/antlr-gemerator/templates/extconf.rb.erb

```diff
@@ -7,6 +7,8 @@ have_library('stdc++')
 
 $CFLAGS << ' -std=c++14'
 
+$defs.push '-DANTLR4CPP_STATIC' unless $defs.include?('-DANTLR4CPP_STATIC')
```

Comment:
Even after reading the [docs](https://github.com/antlr/antlr4/blob/master/doc/cpp-target.md#special-cases-for-this-antlr-target) I'm not sure I understand the need for this flag. Is the idea to generate a static library when building C++ code with Visual Studio? If so, why is that not something you could configure inside Visual Studio itself?

Reply:
It doesn't compile on windows without the static flag.  I added the option to make it optional with a cmd line flag